package/kedro_viz/api/rest/router.py

```diff
@@ -61,7 +61,7 @@ async def get_single_pipeline_data(registered_pipeline_id: str):
 async def deploy_kedro_viz(input_values: DeployerConfiguration):
     try:
         deployer = DeployerFactory.create_deployer(
-            input_values.platform, input_values.endpoint, input_values.bucket_name
+            input_values.platform, input_values.preview,input_values.endpoint, input_values.bucket_name
```

Comment:
since preview is related to the data we construct, it is better to have it at the json level, like in the deploy function -

```python
deployer.deploy(is_preview_enabled)
```
At this moment we only have preview customization for deploy but in future if we have more customizations, we can change the deploy function to accept a DeployDataConfiguration class which contains all the required customizations. For now, we can pass just the boolean.

Reply:
Done.